crates/topos-uci/src/certificate_id.rs

```diff
@@ -69,3 +86,56 @@ impl CertificateId {
         &self.id
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::CertificateId;
+
+    const CERTIFICATE_ID_WITH_PREFIX: &str =
+        "0x11db8713a79c41625f4bb2221bd43ac4766fff23e78f82212f48713a6768e76a";
+    const CERTIFICATE_ID_WITHOUT_PREFIX: &str =
+        "11db8713a79c41625f4bb2221bd43ac4766fff23e78f82212f48713a6768e76a";
+    const MALFORMATTED_CERTIFICATE_ID: &str = "invalid_hex_string";
+
+    #[test]
+    fn convert_cert_id_string_with_prefix() {
+        let certificate_id1: CertificateId = CERTIFICATE_ID_WITH_PREFIX.to_string().into();
+
+        assert_eq!(
+            &certificate_id1.id[..],
+            &[
+                0x11, 0xdb, 0x87, 0x13, 0xa7, 0x9c, 0x41, 0x62, 0x5f, 0x4b, 0xb2, 0x22, 0x1b, 0xd4,
+                0x3a, 0xc4, 0x76, 0x6f, 0xff, 0x23, 0xe7, 0x8f, 0x82, 0x21, 0x2f, 0x48, 0x71, 0x3a,
+                0x67, 0x68, 0xe7, 0x6a
+            ][..]
+        )
+    }
+
+    #[test]
+    fn convert_cert_id_string_without_prefix() {
+        let certificate_id1: CertificateId = CERTIFICATE_ID_WITHOUT_PREFIX.to_string().into();
+
+        assert_eq!(
+            &certificate_id1.id[..],
+            &[
+                0x11, 0xdb, 0x87, 0x13, 0xa7, 0x9c, 0x41, 0x62, 0x5f, 0x4b, 0xb2, 0x22, 0x1b, 0xd4,
+                0x3a, 0xc4, 0x76, 0x6f, 0xff, 0x23, 0xe7, 0x8f, 0x82, 0x21, 0x2f, 0x48, 0x71, 0x3a,
+                0x67, 0x68, 0xe7, 0x6a
+            ][..]
+        )
+    }
+
+    #[test]
+    fn malformatted_cert_id() {
```

Comment:
This should not fallback to `0x000` which is the genesis cert id. If a cert is not parsable, it should return an error